tests/phpunit/tests/html-api/wpHtmlProcessorBreadcrumbs.php

```diff
@@ -0,0 +1,370 @@
+	 * has full HTML5 support. Because of this, these tests lock down the interface
+	 * to ensure that support isn't accidentally updated in one place for a new
+	 * element while overlooked in another.
+	 *
+	 * @ticket 58517
+	 *
+	 * @covers WP_HTML_Processor::step
+	 *
+	 * @dataProvider data_unsupported_elements
+	 *
+	 * @param string $html HTML string containing unsupported elements.
+	 */
+	public function test_fails_when_encountering_unsupported_tag( $html ) {
+		$p = WP_HTML_Processor::createFragment( $html );
+
+		$this->assertFalse( $p->step(), "Should not have stepped into unsupported {$p->get_tag()} element." );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	public function data_unsupported_elements() {
+		$unsupported_elements = array(
+			'ABBR',
+			'ACRONYM', // Neutralized
+			'ADDRESS',
+			'APPLET', // Deprecated
+			'AREA',
+			'ARTICLE',
+			'ASIDE',
+			'AUDIO',
+			'BASE',
+			'BDI',
+			'BDO',
+			'BGSOUND', // Deprecated; self-closing if self-closing flag provided, otherwise normal.
+			'BLINK', // Deprecated
+			'BODY',
+			'BR',
+			'BUTTON',
+			'CANVAS',
+			'CAPTION',
+			'CENTER', // Neutralized
+			'CITE',
+			'COL',
+			'COLGROUP',
+			'DATA',
+			'DATALIST',
+			'DD',
+			'DEL',
+			'DETAILS',
+			'DEFN',
+			'DIALOG',
+			'DL',
+			'DT',
+			'EMBED',
+			'FIELDSET',
+			'FOOTER',
+			'FORM',
+			'FRAME',
+			'FRAMESET',
+			'H1',
+			'H2',
+			'H3',
+			'H4',
+			'H5',
+			'H6',
+			'HEAD',
+			'HEADER',
+			'HGROUP',
+			'HR',
+			'HTML',
+			'IFRAME',
+			'INPUT',
+			'INS',
+			'ISINDEX', // Deprecated
+			'KBD',
+			'KEYGEN', // Deprecated; void
+			'LABEL',
+			'LEGEND',
+			'LI',
+			'LINK',
+			'LISTING', // Deprecated, use PRE instead.
+			'MAIN',
+			'MAP',
+			'MARK',
+			'MARQUEE', // Deprecated
+			'MATH',
+			'MENU',
+			'META',
+			'METER',
+			'MULTICOL', // Deprecated
+			'NAV',
+			'NEXTID', // Deprecated
+			'NOBR', // Neutralized
+			'NOEMBED', // Neutralized
+			'NOFRAMES', // Neutralized
+			'NOSCRIPT',
+			'OBJECT',
+			'OL',
+			'OPTGROUP',
+			'OPTION',
+			'OUTPUT',
+			'PICTURE',
+			'PLAINTEXT', // Neutralized
+			'PRE',
+			'PROGRESS',
+			'Q',
+			'RB', // Neutralized
+			'RP',
+			'RT',
+			'RTC', // Neutralized
+			'RUBY',
+			'SAMP',
+			'SCRIPT',
+			'SECTION',
+			'SELECT',
+			'SLOT',
+			'SOURCE',
+			'SPACER', // Deprecated
+			'SPAN',
+			'STYLE',
+			'SUB',
+			'SUMMARY',
+			'SUP',
+			'SVG',
+			'TABLE',
+			'TBODY',
+			'TD',
+			'TEMPLATE',
+			'TEXTAREA',
+			'TFOOT',
+			'TH',
+			'THEAD',
+			'TIME',
+			'TITLE',
+			'TR',
+			'TRACK',
+			'UL',
+			'VAR',
+			'VIDEO',
+			'WBR',
+			'XMP', // Deprecated, use PRE instead.
+
+			// Made up elements, custom elements.
+			'X-NOT-AN-HTML-ELEMENT',
+			'HUMAN-TIME',
+		);
+
+		$data = array();
+		foreach ( $unsupported_elements as $tag_name ) {
+			$data[ $tag_name ] = array( "<{$tag_name}>" );
+		}
+
+		return $data;
+	}
+
+	/**
+	 * @ticket 58517
```

Comment:
- Needs a test description 🔢 Applies elsewhere in the PR
- Needs an `@covers` annotation here.

Reply:
same question from above: given that that test name itself forms the test description, and that the vast majority of existing tests rely on this, do we really need a duplicated description in every test?